projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -642,6 +642,22 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         }
     }
 
+    /**
+     * @hidden
+    */
+    @HostBinding('style.width')
+    get hostWidth() {
+        if (this.width === null && this.columnList && this.columnList.length > 0) {
+            let width = this.columnList.reduce((sum, item) => sum + parseInt((item.width || item.defaultWidth), 10), 0);
+            if (this.hasVerticalSroll()) {
+                width += this.scrollWidth;
```

Comment:
I think it should add also the features columns width and etc.
Otherwise for example if Row Selection is enabled it will create scrollbar because the column containing checkboxes is not taken into account.